src/components/atom/AppDigitsValidator.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useRef } from "react";
+import DigitsValidatorProps from "../../types/DigitsValidatorProps";
+export const AppDigitsValidator = ({
+                    label,
+                    digitsCount,
+                    value,
+                    onChange,
+                    hasError,
+                    error,
+                    }: DigitsValidatorProps) => {
+    
+    const digitsRef = useRef<HTMLInputElement[]>([]);
+
+
+    const onDigitsChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const {value: fieldValue} = e.target;
+        const isOnlyNumbers = /^\d+$/.test(fieldValue);   
+
+        let insertedValue = (e.nativeEvent as InputEvent).data || '';
+
+
+        if(!isOnlyNumbers) {
+            e.target.value = fieldValue.replace(/[^0-9]/g, '');
```

Comment:
Double-check with the product team if this is a desired behavior. It feels kinda aggressive if you directly delete an input from the user